dashboards-observability/public/components/visualizations/charts/helpers/viz_types.ts

```diff
@@ -20,13 +21,29 @@ interface IVizContainerProps {
   };
 }
 
-const getDefaultXYAxisLabels = (vizFields: IField[]) => {
+const getDefaultXYAxisLabels = (vizFields: IField[], visName: string) => {
   if (isEmpty(vizFields)) return {};
   const vizFieldsWithLabel = vizFields.map(vizField => ({ ...vizField, label: vizField.name }));
-  return {
-    xaxis: [vizFieldsWithLabel[vizFieldsWithLabel.length - 1]] || [],
-    yaxis: take(vizFieldsWithLabel, vizFieldsWithLabel.length - 1 > 0 ? vizFieldsWithLabel.length - 1 : 1) || [],
-  };
+  let xAxis = [];
+  let yAxis = [];
+  const mapXaxis = () => {
+    if (visName === visChartTypes.Line) {
+      xAxis = vizFieldsWithLabel.filter((field) => field.type === 'timestamp')
+    } else {
+      xAxis = [vizFieldsWithLabel[vizFieldsWithLabel.length - 1]]
+    }
+    return xAxis;
+  }
+
+  const mapYaxis = () => {
+    if (visName === visChartTypes.Line) {
+      yAxis = vizFieldsWithLabel.filter((field) => field.type !== 'timestamp')
+    } else {
+      yAxis = take(vizFieldsWithLabel, vizFieldsWithLabel.length - 1 > 0 ? vizFieldsWithLabel.length - 1 : 1) || [];
+    }
+    return yAxis;
+  }
```

Comment:
You do not need these global variables `xAxis` and `yAxis` at all. You could change the function signatures to always return an array. Your functions do not specify any return type.
```JS
const mapXaxis = (): [] => visName === visChartTypes.Line ?
  vizFieldsWithLabel.filter((field) => field.type === 'timestamp') :
  [vizFieldsWithLabel[vizFieldsWithLabel.length - 1]]
```

Reply:
Thanks, Subrat! added the suggested changes and added type too.